lib/spack/spack/extensions.py

```diff
@@ -103,15 +104,14 @@ def get_module(cmd_name):
         cmd_name (str): name of the command for which to get a module
             (contains ``-``, not ``_``).
     """
-    # If built-in failed the import search the extension
-    # directories in order
+
+    pname = spack.cmd.python_name(cmd_name)
     extensions = spack.config.get('config:extensions') or []
-    for folder in extensions:
-        module = load_command_extension(cmd_name, folder)
```

Comment:
Imagine a command "dev," implemented in [dev.py](https://github.com/chissg/spackdev/blob/master/dev/cmd/dev.py), which has sub-commands which are discovered at run time. The following code will _not_ work when loaded with `load_command_extension()` because the package-global variable `_subcmd_functions` populated via calls to `add_subcommand()` is not available to the `dev()` function, so the selected sub-command cannot be invoked. This problem is resolved if `dev.py` is imported rather than loaded.

Reply:
"not available" is a little sloppy: the _value_ initialized by one function is not available to the other.